src/compiler/binder.ts

```diff
@@ -2230,6 +2238,36 @@ namespace ts {
             declareSymbol(funcSymbol.members, funcSymbol, leftSideOfAssignment, SymbolFlags.Property, SymbolFlags.PropertyExcludes);
         }
 
+        function bindPropertyAssignment(node: BinaryExpression) {
+            // We saw a node of the form 'x.y = z'. Declare a 'member' y on x if x was a function.
+
+            // Look up the function in the local scope, since prototype assignments should
+            // follow the function declaration
+            const leftSideOfAssignment = node.left as PropertyAccessExpression;
+            const target = leftSideOfAssignment.expression as Identifier;
+
+            // Fix up parent pointers since we're going to use these nodes before we bind into them
+            leftSideOfAssignment.parent = node;
+            target.parent = leftSideOfAssignment;
+
+            let funcSymbol = container.locals.get(target.text);
+            if (funcSymbol && isDeclarationOfFunctionOrClassExpression(funcSymbol)) {
+                funcSymbol = (funcSymbol.valueDeclaration as VariableDeclaration).initializer.symbol;
+            }
+
+            if (!funcSymbol || !(funcSymbol.flags & (SymbolFlags.Function | SymbolFlags.Class))) {
+                return;
+            }
+
+            // Set up the members collection if it doesn't exist already
+            if (!funcSymbol.exports) {
+                funcSymbol.exports = createMap<Symbol>();

```

Comment:
Is it because this is the static side of the class, not the instance side? Exports is a confusing name in this case, then.